### DISCUSSION TITLE
Cannot access Kibana on 2.3.181 getting 404 error and 502's for Alerts and Dashboards

### DISCUSSION THREAD

**Discussion by kgoode517** [Security-Onion-Solutions/securityonion#9025] [Unsupported Versions] [answered]
Need assistance interpreting docker logs for kibana on previously working standalone server seeing kibana errors but not sure how to resolve them attempted solutions for #8335, #8399, and #8435 to no avail. Issue seems to have occurred after a soup update and I cannot resolve. so-suricata and so-status look good. Docker ps shows all containers running current version of 2.3.181, opt/so/log.soc/ logs show no errors. 
[kibanadockerlogs.txt](https://github.com/Security-Onion-Solutions/securityonion/files/9894732/kibanadockerlogs.txt)

**Comment by rosswakelin** [comment]
I see this after about 50% of the soup updates.  I fix it by rebooting the database servers in the cluster (one at a time of course)

**Reply by kgoode517** [reply]
@rosswakelin how would you suggest doing that? I have attempted to restart the onion or even clear the database with so-elastic-clear but still getting the same errors.

**Comment by rosswakelin** [comment]
Hmm, don't know what you have done to your systems, but here we just ssh into each database box and perform a `sudo shutdown -r n`

**Comment by dougburks** [comment]
@kgoode517 Looking at your kibanadockerlogs.txt, it appears that Kibana was having difficulty querying Elasticsearch. If your machine is underpowered, then it could take a while for Elasticsearch to fully initialize so you might want to review the hardware requirements for a standalone deployment at https://docs.securityonion.net/en/2.3/hardware.html#standalone-deployments. 

If your machine has sufficient hardware, then the next thing I would do is check the Elasticsearch log for additional clues. If all else fails, you can always perform a fresh installation.

**Reply by kgoode517** [reply]
@dougburks thanks for evaluating. My hardware is sufficient. Elasticsearch log shows no errors. Fresh install is usually my go to when this happens but is there anything else you would recommend to try before resorting to that and having to reset everything up again?

**Reply by dougburks** [reply]
> My hardware is sufficient.

What exactly are your hardware specs?

> Elasticsearch log shows no errors.

What Elasticsearch log did you look at?

>  Fresh install is usually my go to when this happens but is there anything else you would recommend to try before resorting to that and having to reset everything up again?

You could troubleshoot this for days or weeks, but it might be quicker and easier to do a fresh install.

**Comment by kgoode517** [comment, marked as answer]
6 cpu core, 16 gb ram, 200 gig disk space on Virtual Box Security Onion.  sudo docker logs so-elasticsearch. You are of course right fresh install saves a bunch of time I just hate that I have to do that so often, set all the clients back up and lose previous saved cases.  I should be checking my alerts everyday to catch this quicker though. I can confirm it is working fine on the fresh install.

**Reply by dougburks** [reply]
> 6 cpu core, 16 gb ram, 200 gig disk space

Those specs are pretty close to the MINIMUM requirements stated at https://docs.securityonion.net/en/2.3/hardware.html#standalone-deployments. Depending on how much traffic you're monitoring and how many clients are sending logs, you will most likely need more horsepower.

>  on Virtual Box

This could be part of the problem, especially if there is resource contention with other VMs. Please see https://docs.securityonion.net/en/2.3/hardware.html#virtualization and https://docs.securityonion.net/en/2.3/best-practices.html.

> sudo docker logs so-elasticsearch

You likely won't see much here. Most Elasticsearch logs are stored in the filesystem as shown at https://docs.securityonion.net/en/2.3/elasticsearch.html#diagnostic-logging.

> You are of course right fresh install saves a bunch of time I just hate that I have to do that so often

You shouldn't have to do that often, if at all, with a properly configured deployment.

**Comment by kgoode517** [comment]
Ah was following the recommendation of at least 12 gb ram for a general deployment now seeing in  Stand Alone that 16 is the minimum requirement and would likely need swap space to not cause issues. I added additional ram. This was likely the issue after upgrading to Elastic 8 needing more resources that were already somewhat constrained. Also checked /opt/so/log/elasticsearch at the time and did not see anything.  Another contributing factor to this issue in the past could likely be using the power off function instead of the send shutdown signal for Virtual Box when upgrading the os of the host machine or doing routine maintenance.
